src/main/java/lt/vu/application/order/service/DiscountedPriceCalculator.java

```diff
@@ -0,0 +1,20 @@
+package lt.vu.application.order.service;
+
+import lt.vu.persistence.orm.entities.Order;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Specializes;
+
+@Alternative
+@Specializes
```

Comment:
`@Specializes` means this one is being picked by default over other implementations. `@Alternative` only gets enabled when we specify it in `beans.xml` file. Right now my guess is that event if you remove alternative from `beans.xml`, `DiscountedPriceCalculator` will still be used because of `@Specializes`.

Reply:
I'm a bit confused with `@Specializes`. 

> Usually, a bean marked with the `@Specializes` annotation is also an alternative and is declared as an alternative in the beans.xml file. Such a bean is meant to stand in as a replacement for the default implementation, and the alternative implementation automatically inherits all qualifiers of the default implementation as well as its EL name, if it has one.

https://docs.oracle.com/javaee/6/tutorial/doc/gjsdf.html#gkhpo

So, I guess it works as `@Alternative`, but it also overrides other alternatives(?). I did it just for the sake of requirement, probably it could've been only `@Alternative` in this case.

Besides that, it works fine when you remove from `beans.xml`.

Reply:
I did some testing, seems like removing `alternative` from `beans.xml` makes the default calculator work again 🤷‍♂️